src/utils/text_processing.py

```
@@ -359,34 +314,16 @@ def extract_keywords(text: str, top_n: int = 5) -> list[str]:
         unwanted = get_unwanted_words()
         words = [w for w in words if w not in unwanted]
-
-        # Get frequency distribution
-        freq_dist = FreqDist(words)
-
-        # Return top N
-        return [word for word, _ in freq_dist.most_common(top_n)]
-
-    except (LookupError, ValueError, OSError, RuntimeError) as e:
+        return [word for word, _ in Counter(words).most_common(top_n)]
+    except Exception as e:
```

Ruff (0.15.10)

[warning] Do not catch blind exception: `Exception`

(BLE001)